src/irr/asset/CGLTFLoader.cpp

```diff
@@ -34,16 +35,427 @@ namespace irr
 
 		asset::SAssetBundle CGLTFLoader::loadAsset(io::IReadFile* _file, const asset::IAssetLoader::SAssetLoadParams& _params, asset::IAssetLoader::IAssetLoaderOverride* _override, uint32_t _hierarchyLevel)
 		{
+			CGLTFHeader header;
 			simdjson::dom::parser parser;
 			simdjson::dom::object tweets = parser.load(_file->getFileName().c_str());
 			simdjson::dom::element element;
 
+			std::filesystem::path filePath(_file->getFileName().c_str());
+			const std::string rootAssetDirectory = std::filesystem::absolute(filePath.remove_filename()).u8string();
+
+			constexpr uint8_t POSITION_ATTRIBUTE = 0;
+			constexpr uint8_t NORMAL_ATTRIBUTE = 3;
+
+			auto mesh = core::make_smart_refctd_ptr<CCPUMesh>();
+			auto mbVertexShader = core::smart_refctd_ptr<ICPUSpecializedShader>();
+			auto mbFragmentShader = core::smart_refctd_ptr<ICPUSpecializedShader>();
+
 			for (auto& [key, value] : tweets)
 			{
+				auto meshbuffer = core::make_smart_refctd_ptr<ICPUMeshBuffer>();
+				meshbuffer->setPositionAttributeIx(POSITION_ATTRIBUTE);
+				meshbuffer->setNormalnAttributeIx(NORMAL_ATTRIBUTE);
+
 				if (key == "asset")
 				{
 					tweets.at_key("asset").at_key("version").get(element);
 					header.version = std::stoi(element.get_string().value().data());
+
+					auto& minVersion = value.at_key("minVersion");
+					if (minVersion.error() != simdjson::error_code::NO_SUCH_FIELD)
+					{
+						header.minVersion = minVersion.get_uint64().value();
+						if (header.minVersion.value() > header.version)
+							return {};
+					}
+
+					auto& generator = value.at_key("generator");
+					if (generator.error() != simdjson::error_code::NO_SUCH_FIELD)
+						header.generator = generator.get_string().value().data();
+
+					auto& copyright = value.at_key("copyright");
+					if (copyright.error() != simdjson::error_code::NO_SUCH_FIELD)
+						header.copyright = copyright.get_string().value().data();
+				}
+
+				/*
+					Buffers and buffer views do not contain type information.
+					They simply define the raw data for retrieval from the file.
+					Objects within the glTF file (meshes, skins, animations) access buffers
+					or buffer views via accessors.
+				*/
+
+				else if (key == "buffers")
+				{
+					for (auto& buffer : value)
+					{
+						auto& byteLength = value.at_key("byteLength");
+						if (byteLength.error() != simdjson::error_code::NO_SUCH_FIELD)
+						{
+							auto byteLengthVal = byteLength.get_uint64().value();
+							if (byteLengthVal < 1)
+								continue;
+						}
+						else
+							continue;
+
+						auto& uri = value.at_key("uri");
+						if (uri.error() != simdjson::error_code::NO_SUCH_FIELD)
+						{
+							std::string_view uriBin = byteLength.get_string().value(); // TODO
+
+							const asset::IAssetLoader::SAssetLoadParams params;
+							auto buffer_bundle = assetManager->getAsset(rootAssetDirectory + uriBin.data(), params);
```

Comment:
dont add the directory to the URI, the loader overload (default one) should figure that out AFAIK (see how OBJ and others do it)

Reply:
btw in the params you can request that only a buffer asset get loaded.

Reply:
which is safer for the static cast that takes place later.